lib/index.js

```diff
@@ -65,28 +65,76 @@ exports.runOperation = async ({operation}) => {
     }
   }
 
-  if(isGenerateKeyOp) {
-    // TODO: handle cornercase where record insert worked previously but the
-    // plugin did not generate the key... currently, we insert the record
-    // first in case we crash before using the plugin but perhaps this should
-    // be inverted once this TODO is handled
+  // FIXME: remove comments
+  /*
+  client generates key ->
+
+  1. key pending key record is created in bedrock-kms
+  <failure> ssm request to generate key is never made
+  result, pending record in kms and no key record in ssm
+  state:
+    kms -> pending key
+    ssm -> no key
+
+  2. key pending key record is created in bedrock-kms
+  ssm request to generate key succeeds
+  <failure> kms does not record successful key generation (remove pending flag)
+  state:
+    kms -> pending key
+    ssm -> proper key
+
+  3. key pending key record is created in bedrock-kms
+  ssm request to generate key fails
+  key record is removed from bedrock-kms
```

Comment:
Presently, kms key records are not being removed, they are just left as pending.  I think this is fine.

Reply:
Yeah, I can accept that.